packages/cli/generation/ir-generator/src/examples/generator/generatePrimitiveExample.ts

```diff
@@ -74,3 +89,110 @@ export function generatePrimitiveExample({
             assertNever(primitiveType.v1);
     }
 }
+
+function maybeStringValidation(v2: PrimitiveTypeV2 | undefined): StringValidationRules | undefined {
+    if (v2?.type === "string") {
+        const stringType = v2 as PrimitiveTypeV2.String;
+        return stringType.validation;
+    }
+    return undefined;
+}
+
+function maybeDoubleValidation(v2: PrimitiveTypeV2 | undefined): DoubleValidationRules | undefined {
+    if (v2?.type === "double") {
+        const doubleType = v2 as PrimitiveTypeV2.Double;
+        return doubleType.validation;
+    }
+    return undefined;
+}
+
+function maybeIntegerValidation(v2: PrimitiveTypeV2 | undefined): IntegerValidationRules | undefined {
+    if (v2?.type === "integer") {
+        const integerType = v2 as PrimitiveTypeV2.Integer;
+        return integerType.validation;
+    }
+    return undefined;
+}
+
+function getStringExampleOfLength(length: number): string {
+    if (length <= Examples.SAMPLE_STRINGS.length) {
+        const sampleString = Examples.SAMPLE_STRINGS[length - 1];
+        if (sampleString) {
+            return sampleString;
+        } else {
+            throw new Error(`Unexpected undefined value in SAMPLE_STRINGS at index ${length - 1}`);
+        }
+    }
+    return (
+        Examples.SAMPLE_STRINGS[Examples.SAMPLE_STRINGS.length - 1] +
+        ".".repeat(length - Examples.SAMPLE_STRINGS.length)
+    );
+}
+
+function generatePrimitiveStringExample({
+    fieldName,
+    validation
+}: {
+    fieldName: string | undefined;
+    validation: StringValidationRules | undefined;
+}): ExampleGenerationSuccess<ExamplePrimitive> {
+    if (validation) {
+        const minLength = validation.minLength;
+        const maxLength = validation.maxLength;
+        if (maxLength) {
+            const maxLengthExample = getStringExampleOfLength(maxLength);
+            return {
+                type: "success",
+                example: ExamplePrimitive.string({ original: maxLengthExample }),
+                jsonExample: maxLengthExample
+            };
+        } else if (minLength) {
+            const minLengthExample = getStringExampleOfLength(minLength);
+            return {
+                type: "success",
+                example: ExamplePrimitive.string({ original: minLengthExample }),
+                jsonExample: minLengthExample
+            };
+        }
```

Comment:
Let's swap the order here - we should prefer the minimum whenever possible. And in the case that we only have a maxmium and no minimum, we should prefer a length that is strictly smaller than the maximum. Otherwise users will see examples with a ton of content (e.g. 255 characters).

Reply:
Done!